### PR TITLE
fix: switch to tf.random.Generator

### LINKED ISSUE #139: Write test to check that UniformRealNumberGenerator generates uniform numberes

### LINKED ISSUE #140: Replace tf.random.uniform with tf.random.Generator
Since upgrading to TF2.3 (65fc6a00), there's a bug in the phase space generation (found by @sebastianJaeger @Leongrim). The bug can be reproduced with the snippet below (run with e9cc6b6). The issue seems to be caused by `tf.random.uniform` (see https://github.com/ComPWA/tensorwaves/issues/59#issuecomment-716478566 and comment in the code below) and can be avoided by switching to `tf.random.Generator`.

```python
import matplotlib.pyplot as plt
import pandas as pd
import tensorflow as tf
from expertsystem.amplitude import generate_amplitude_model
from expertsystem.reaction import InteractionTypes, StateTransitionManager

from tensorwaves.data.tf_phasespace import TFPhaseSpaceGenerator
from tensorwaves.physics.helicity_formalism.kinematics import (
    HelicityKinematics,
)

stm = StateTransitionManager(
    initial_state=[("J/psi(1S)", [-1, +1])],
    final_state=["gamma", "pi0", "pi0"],
    allowed_intermediate_particles=["f(0)(980)"],
    formalism_type="helicity",
)
stm.set_allowed_interaction_types([InteractionTypes.Strong])
graph_interaction_settings_groups = stm.prepare_graphs()
result = stm.find_solutions(graph_interaction_settings_groups)
model = generate_amplitude_model(result)

kinematics = HelicityKinematics.from_model(model)
kinematics.register_invariant_mass([3, 4])
kinematics.register_invariant_mass([2, 3])
phsp_gen_instance = TFPhaseSpaceGenerator(kinematics.reaction_kinematics_info)
tf.random.set_seed(12345)  # <-- remove to avoid the bug
data_sample, weights = phsp_gen_instance.generate(30000, None)  # type: ignore
dataset = kinematics.convert(data_sample)
data_frame = pd.DataFrame(dataset)
plt.hist2d(
    data_frame["mSq_3_4"],
    data_frame["mSq_2_3"],
    bins=(100, 100),
    cmin=1e-6,
)
plt.show()
```

### PR DESCRIPTION
**Pull request by redeboer**
Closes #139
Closes #140

### CODE REVIEW

File: tensorwaves/data/generate.py [hunk 1]
```diff
@@ -51,9 +51,9 @@ def generate_data(
         [ParticleReactionKinematicsInfo], PhaseSpaceGenerator
     ] = TFPhaseSpaceGenerator,
     random_generator: Callable[
-        [float], UniformRealNumberGenerator
+        [int], UniformRealNumberGenerator
     ] = TFUniformRealNumberGenerator,
-    seed: float = 123456.0,
+    seed: int = 123456,
```
Comment (spflueger) [inline tensorwaves/data/generate.py:56]: Maybe its even best to remove the seed and leave that responsibility to the random_generator instance.

Comment (redeboer) [reply tensorwaves/data/generate.py:56]: Yeah good point

**Review by Leongrim — APPROVED** [review]

File: tensorwaves/data/generate.py (hunk 1 above)
Comment (redeboer) [reply tensorwaves/data/generate.py]: Hope 39dcde7 is what you had in mind

Comment (spflueger) [reply tensorwaves/data/generate.py]: looks good

**Review by spflueger — APPROVED** [review]
GJ

**Pull request merged**